File: docker/docker_mng.py
#!/usr/bin/env python3.5

import os
from subprocess import run, PIPE
import sys

IMAGE = 'nlan/agent:ver0.1'  # Docker image name

def _set_symbolic_link(node):
    run(['mkdir', '-p', '/var/run/netns'])
    pid = run(['docker', 'inspect', '--format', '{{.State.Pid}}', node], stdout=PIPE, universal_newlines=True).stdout.rstrip()
    target = '/var/run/netns/' + node
    if os.path.islink(target):
        os.remove(target)
    run(['ln', '-s', '/proc/{}/ns/net'.format(pid), target])
    print('pid ' + pid)

def _run(node):
    run(['docker', 'run', '-i', '-t', '-d', '-v', '/tmp:/var/volume:rw', '--privileged', '--name', node, '--env', 'HOSTNAME='+node, IMAGE])
    _set_symbolic_link(node)

def _rm(node):
    run(['docker', 'rm', node])

def _start(node):
    run(['docker', 'start', node])
    _set_symbolic_link(node)

def _stop(node):
    run(['docker', 'stop', node])

if __name__ == '__main__':
    ope = sys.argv[1]
    nodes = sys.argv[2:]
    for node in nodes:
        globals()['_'+ope](node)

File: docker/docker_mng.go
package main

import (
	"fmt"
	"os"
	"os/exec"
)

const (
	IMAGE = "nlan/agent:ver0.1" // Docker image name
)

func main() {
	ope := os.Args[1]
	nodes := os.Args[2:]
	var cmd func(string) ([]byte, error)
	switch ope {
	case "run":
		cmd = func(node string) ([]byte, error) {
			arg := []string{"run", "-i", "-t", "-d", "-v", "/tmp:/var/volume:rw", "--privileged", "--name", node, "--env", "HOSTNAME=" + node, IMAGE}
			return exec.Command("docker", arg...).Output()
		}
	case "rm":
		cmd = func(node string) ([]byte, error) {
			return exec.Command("docker", "rm", node).Output()
		}
	case "start":
		cmd = func(node string) ([]byte, error) {
			return exec.Command("docker", "start", node).Output()
		}
	case "stop":
		cmd = func(node string) ([]byte, error) {
			return exec.Command("docker", "stop", node).Output()
		}
	default:
		fmt.Println("Operation unidentified")
		os.Exit(1)
	}
	for _, node := range nodes {
		out, err := cmd(node)
		fmt.Printf("%s", out)
		if err != nil {
			fmt.Println("Operation failure")
			fmt.Println(err)
			os.Exit(1)
		}
	}
}
